crates/typst-cli/src/compile.rs

```diff
@@ -118,7 +119,9 @@ pub fn compile_once(
                 .map_err(|err| eco_format!("failed to print diagnostics ({err})"))?;
 
             if let Some(open) = command.open.take() {
-                open_file(open.as_deref(), &command.output())?;
+                if let Output::Path(file) = command.output() {
+                    open_file(open.as_deref(), &file)?;
+                }
```

Comment:
What's the better way to show a warning here about opening from Stdout?

Reply:
The closest option is `conflict_with`, but it cannot compare values like `required_if_eq` can. I added docs for --open.